File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务接口
 */
public interface BrandService {
    /*
      *简单的查询所有的品牌信息
     */
   List<TbBrand> findAll();

    /**
     *品牌分页
     * pageNum  当前记录
     * pageSize 每页的记录数
     */
    PageResult findPage(int pageNum,int pageSize);

    /**
     * 添加品牌
     */
   Exception add(TbBrand tbBrand);

    /**
     * 根据ID查询实体信息
     */
    TbBrand findOne(Long id);

    /**
     * 修改信息
     */
    void update(TbBrand tbBrand);

    /**
     * 删除信息
     * Long[] ids
     */
    void delete(Long[] ids);

    /**
     * 更加条件,关键子,分页查询数据
     * brand , pageNume, pageSize
     */
    PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);

    /**
     *  品牌下拉列表展示
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务层
app.service('brandService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../brand/findAll.action');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../brand/findPage.action?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../brand/findOne.action?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../brand/addBrand.action',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../brand/updateBrand.action',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../brand/deleteBrand.action?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../brand/searchBrand.action?page='+page+"&rows="+rows, searchEntity);
	}    
	//品牌下拉列表数据
	this.selectOptionList=function(){
		return $http.get('../brand/selectOptionList.action');
	}
	
});
